helper/resource/testing_new.go

```diff
@@ -0,0 +1,615 @@
+	state := terraform.NewState()
+	state.TFVersion = jsonState.TerraformVersion
+	if jsonState.Values == nil {
+		// the state is empty
+		return state, nil
+	}
+
+	if err := shimStateModule(state, jsonState.Values.RootModule); err != nil {
+		return nil, err
+	}
+
+	// shimming of lists and maps might be incorrect
+	for key, output := range jsonState.Values.Outputs {
+		outputType := ""
+		switch output.Value.(type) {
+		case string:
+			outputType = "string"
+		case []interface{}:
+			outputType = "list"
+		case map[string]interface{}:
+			outputType = "map"
+		default:
+			return nil, errors.New("output was not expected type")
+		}
+
+		state.RootModule().Outputs[key] = &terraform.OutputState{
+			Type:      outputType,
+			Value:     output.Value,
+			Sensitive: output.Sensitive,
+		}
+	}
+
+	return state, nil
+}
+
+func getState(t *testing.T, wd *tftest.WorkingDir) *terraform.State {
+	jsonState := wd.RequireState(t)
+	state, err := shimTFJson(jsonState)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return state
+}
+
+func RunLegacyTest(t *testing.T, c TestCase, providers map[string]terraform.ResourceProvider) {
+	spewConf := spew.NewDefaultConfig()
+	spewConf.SortKeys = true
+	wd := acctest.TestHelper.RequireNewWorkingDir(t)
+
+	defer func() {
+		// destroy step
+		// TODO probably better to combine this with TestStep.Destroy implementation as in old framework
+
+		wd.RequireDestroy(t)
+
+		if c.CheckDestroy != nil {
+			statePostDestroy := getState(t, wd)
+
+			if err := c.CheckDestroy(statePostDestroy); err != nil {
+				t.Fatal(err)
+			}
+		}
+		wd.Close()
+	}()
+
+	providerCfg := testProviderConfig(c)
+
+	wd.RequireSetConfig(t, providerCfg)
+	wd.RequireInit(t)
+
+	// use this to track last step succesfully applied
+	// acts as default for import tests
+	var appliedCfg string
+
+	for i, step := range c.Steps {
+
+		if step.PreConfig != nil {
+			step.PreConfig()
+		}
+
+		// TODOs
+		if step.Destroy {
+			t.Fatal("TODO: TestStep.Destroy")
+		}
+
+		if step.SkipFunc != nil {
+			skip, err := step.SkipFunc()
+			if err != nil {
+				t.Fatal(err)
+			}
+			if skip {
+				log.Printf("[WARN] Skipping step %d", i)
+				continue
+			}
+		}
+
+		if step.ImportState {
+			step.providers = providers
+			err := testStepNewImportState(t, c, wd, step, appliedCfg)
+			if err != nil {
+				t.Fatal(err)
+			}
+			continue
+		}
+
+		if step.Config != "" {
+			err := testStepNewConfig(t, c, wd, step)
+			if step.ExpectError != nil {
+				if err == nil {
+					t.Fatal("Expected an error but got none")
+				}
+				if !step.ExpectError.MatchString(err.Error()) {
+					t.Fatalf("Expected an error with pattern, no match on: %s", err)
+				}
+			} else {
+				if err != nil {
+					t.Fatal(err)
+				}
+			}
+			appliedCfg = step.Config
+			continue
+		}
+
+		t.Fatal("Unsupported test mode")
+	}
+	// If we never checked an id-only refresh, it is a failure.
+	// TODO KEM: why is this here? does this ever happen?
+	// if idRefresh {
```

Comment:
Do we know if any of "the big providers" even use this? What value does it provide, I'm not sure it's a modern practice I would be okay if we made providers "delete" those tests or rewrite them?

Reply:
Copying some data here from a conversation elsewhere:

There are 806 occurrences of ID refresh tests in `terraform-providers` providers:
```
$ grep -r --exclude-dir="vendor" --include="*_test.go" "IDRefresh" * | wc -l
806
```

and more than half of these are in "big" providers:

```
$ grep -r --exclude-dir="vendor" --include="*_test.go" "IDRefresh" * | grep -e aws -e azure -e google -e kubernetes -e oracle | wc -l
454
```